source/franz.go

```diff
@@ -65,12 +71,25 @@ func NewFranzConsumer(ctx context.Context, cfg Config) (*FranzConsumer, error) {
 		return nil, fmt.Errorf("failed to create kafka client: %w", err)
 	}
 
-	return &FranzConsumer{
+	consumer := &FranzConsumer{
 		client:               cl,
-		acker:                newBatchAcker(cl, 1000),
+		acker:                newBatchAcker(cl, ackBatchSize),
 		iter:                 &kgo.FetchesRecordIter{}, // empty iterator is done
 		retryGroupJoinErrors: cfg.RetryGroupJoinErrors,
-	}, nil
+	}
+
+	go consumer.scheduleFlushing(ctx)
+
+	return consumer, nil
+}
+
+func (c *FranzConsumer) scheduleFlushing(ctx context.Context) {
+	for range time.Tick(ackBatchTime) {
+		err := c.acker.Flush(ctx)
```

Comment:
This is not safe to call in a goroutine right now, we're checking [`curBatchIndex`](https://github.com/ConduitIO/conduit-connector-kafka/blob/120e04ad2fdd35cbd30eb5c27791ccd0daafd12f/source/franz.go#L169-L171) while at the same time it could be incremented in [`Ack`](https://github.com/ConduitIO/conduit-connector-kafka/blob/120e04ad2fdd35cbd30eb5c27791ccd0daafd12f/source/franz.go#L160). It was safe before, because the only place it was called is `Ack` and that's only happening once at a time.

Reply:
Yup, the tests already complained.:) Fixed.